crates/core/component/fee/src/gas.rs

```diff
@@ -55,22 +59,16 @@ pub struct GasPrices {
 
 impl GasPrices {
     pub fn zero() -> Self {
-        Self {
-            block_space_price: 0,
-            compact_block_space_price: 0,
-            verification_price: 0,
-            execution_price: 0,
-        }
+        Self::default()
     }
 
-    /// Calculates the price based on given gas schedule. Applies an implicit
-    /// denominator of 1,000 to the gas prices.
-    pub fn price(&self, gas: &Gas) -> Amount {
+    /// Use these gas prices to calculate the fee for a given gas vector.
+    pub fn fee(&self, gas: &Gas) -> Amount {
         Amount::from(
-            self.block_space_price * (gas.block_space * 1_000) / 1_000
-                + self.compact_block_space_price * (gas.compact_block_space * 1_000) / 1_000
-                + self.verification_price * (gas.verification * 1_000) / 1_000
-                + self.execution_price * (gas.execution * 1_000) / 1_000,
```

Comment:
Since the motivation for the denominator is to allow more precision in prices, we want to apply the multiple to the parameters we eventually use, rather than doing the multiplication here.